pkg/codegen/docs/gen.go

```diff
@@ -1959,10 +1930,69 @@ func (mod *modContext) genIndex() indexData {
 
 	// If this is the root module, write out the package description.
 	if mod.mod == "" {
-		data.PackageDescription = mod.pkg.Description()
+		idxData.PackageDescription = mod.pkg.Description()
 	}
 
-	return data
+	if err := addFileTemplated("", "index.tmpl", idxData); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// indexEntry represents an individual entry on an index page.
+type indexEntry struct {
+	Link        string
+	DisplayName string
+}
+
+// indexData represents the index file data to be rendered as _index.md.
+type indexData struct {
+	Tool string
+
+	Title              string
+	TitleTag           string
+	PackageDescription string
+
+	Functions      []indexEntry
+	Resources      []indexEntry
+	Modules        []indexEntry
+	PackageDetails packageDetails
+}
+
+// indexEntrySorter implements the sort.Interface for sorting
+// a slice of indexEntry struct types.
+type indexEntrySorter struct {
+	entries []indexEntry
+}
```

Comment:
I know this is from before but two things:
When this was needed, it didn't need to be a struct.
It is no longer needed: sort.Slice exists.